src/endpoints/users.js

```diff
@@ -22,6 +22,24 @@ const visibleGroups = (user) => ALL_GROUPS
   }));
 
 
+/**
+ * Returns an array of Nextstrain groups that a given *user* is a member of. The
+ * order of groups returned matches the order in the `groups.json` data file.
+ *
+ * @param {Object | undefined} user. `undefined` represents a non-logged-in user
+ * @returns {Array} Each element represents a group with a a subset of properties from
+ *                  the Group class.
+ */
+const groupMemberships = (user) => user?.groups
+  ?.map(name => new Group(name))
```

Comment:
This ended up breaking `/whoami` for me (noticed in production just now) because my Cognito user is a member of Cognito groups for which there are no corresponding Nextstrain groups. This is a historical artifact and likely only affects me (and maybe one or two other devs), but it'd still be better to be robust in the face of unexpected input.

I noticed the issue in the web, where `/whoami` 404s for me when JSON is requested, but reproduced it below with `curl` for an example:

```console
$ nextstrain whoami
# Logged into https://nextstrain.org as…
username: trs
email: tsibley@fredhutch.org
groups: 
  - !flags/canary
  - blab-private/editors
  - blab/editors
  - seattleflu/editors
  - test/editors
  - trs-test/owners

$ curl -s https://nextstrain.org/whoami -H 'Accept: application/json' -H @<(nextstrain authorization)
{"error":"unknown group: test"}
```

Reply:
I'd [previously said](https://github.com/nextstrain/nextstrain.org/pull/780#discussion_r1448077922):

> Longer term, it probably makes sense to convert `req.user.groups` to Group objects in the authn code rather than here. There can also be some additional error/edge-case handling (e.g. group names reported by Cognito that the server doesn't know about).

and I guess I shouldn't have said "Longer term"!

Reply:
Ahh, fixing in #839